File: 프로그래머스/0/181921. 배열 만들기 2/배열 만들기 2.py
def solution(l, r):
    answer = []
    
    for i in range(l, r+1):
        if all(num in ['0', '5'] for num in str(i)): 
            # 'i'의 모든 자리수가 '0' 또는 '5'으로만 이루어져 있을 때만 참을 반환
            # all(): 모든 요소가 참이면 참을 반환하는 파이썬 내장 함수
            answer.append(i)
        else:
            continue
            
    if len(answer) == 0:
        answer.append(-1)
                
    return answer

File: 프로그래머스/0/181921. 배열 만들기 2/배열 만들기 2.java
import java.util.*;

class Solution {
    public int[] solution(int l, int r) {
        ArrayList<Integer> answer = new ArrayList<>();
        
        for(int i=l; i<=r; i++){
            String i_str = Integer.toString(i);
            boolean valid = true;
            
            for(int j=0; j<i_str.length(); j++){
                char ch = i_str.charAt(j);
                if(ch != '5' && ch != '0'){ // "0"과 "5" 외의 숫자가 있는지 확인
                    valid = false;
                    break;
                } 
            }
            if(valid == true){
                answer.add(i);
            }
        }
        
        // 조건을 만족하는 숫자가 없으면 -1을 반환
        if(answer.isEmpty()){
            return new int[]{-1};
        }
        else{
            return answer.stream().mapToInt(i -> i).toArray();
        }
        
    }
}